tests/scripts/thread-cert/Cert_5_1_02_ChildAddressTimeout.py

```diff
@@ -79,79 +105,74 @@ def test(self):
 
+        self.nodes[MED].stop()
         self.nodes[SED].stop()
-        self.simulator.go(5)
-        for addr in sed_addrs:
-            if addr[0:4] != 'fe80':
-                self.assertFalse(self.nodes[LEADER].ping(addr))
-
-        leader_messages = self.simulator.get_messages_sent_by(LEADER)
-        router1_messages = self.simulator.get_messages_sent_by(ROUTER)
-        ed_messages = self.simulator.get_messages_sent_by(ED)
-        sed_messages = self.simulator.get_messages_sent_by(SED)
+        self.simulator.go(config.DEFAULT_CHILD_TIMEOUT)
+        self.assertFalse(self.nodes[LEADER].ping(med_mleid))
+        self.assertFalse(self.nodes[LEADER].ping(sed_mleid))
 
-        # 1 - All
-        leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
 
-        router1_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        leader_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
+        LEADER = pv.vars['LEADER']
+        ROUTER = pv.vars['ROUTER']
+        MED = pv.vars['MED']
+        SED = pv.vars['SED']
 
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
+        # Step 1: Verify topology is formed correctly
 
-        msg = router1_messages.next_coap_message("0.02")
-        msg.assertCoapMessageRequestUriPath("/a/as")
+        pv.verify_attached('ROUTER')
 
-        msg = leader_messages.next_coap_message("2.04")
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_wpan_dst64(MED).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
 
-        router1_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_wpan_dst64(SED).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
 
-        ed_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-        ed_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
+        # Step 2: Power off both devices and allow for the keep-alive timeout to expire
+        # Step 3: The Leader sends an ICMPv6 Echo Request to MED and attempts to perform
+        #         address resolution by sending an Address Query Request
 
-        sed_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
-        sed_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
+        pkts.filter_wpan_src64(LEADER).\
+            filter_RLARA().\
+            filter_coap_request(ADDR_QRY_URI).\
+            must_next()
```

Comment:
better to check the target tlv in `a/aq` has value of med's mleid?

Reply:
added